template.go

```diff
@@ -57,7 +59,9 @@ func (t *template) Set(name string, val interface{}, attributes ...PropertyAttri
 }
 
 func (t *template) finalizer() {
-	C.TemplateFree(t.ptr)
+	// Using v8::PersistentBase::Reset() wouldn't be thread-safe to do from
+	// this finalizer goroutine so just free the wrapper and let the template
+	// itself get cleaned up when the isolate is disposed.
+	C.TemplateFreeWrapper(t.ptr)
 	t.ptr = nil
-	runtime.SetFinalizer(t, nil)
```

Comment:
From https://pkg.go.dev/runtime#SetFinalizer

> When the garbage collector finds an unreachable block with an associated finalizer, it clears the association and runs finalizer(obj) in a separate goroutine. This makes obj reachable again, but now without an associated finalizer.

so Go should will clear the finalizer for us, thus we don't need to do that here.